digest/src/buffer_macros/fixed.rs

```diff
@@ -39,6 +39,21 @@ macro_rules! buffer_fixed {
             const OID: $crate::const_oid::ObjectIdentifier =
                 $crate::const_oid::ObjectIdentifier::new_unwrap($oid);
         }
+
+        impl$(< $( $lt $( : $clt $(+ $dlt )* )? ),+ >)? Drop for $name$(< $( $lt ),+ >)? {
+            #[inline]
+            fn drop(&mut self) {
+                #[cfg(feature = "zeroize")]
+                {
+                    use $crate::zeroize::Zeroize;
+                    self.core.zeroize();
+                    self.buffer.zeroize();
```

Comment:
I don't think we need this explicit `Drop` impl. You just need to verify that both `core` and `buffer` implement `ZeroizeOnDrop`. It would also make the `decompose` changes unnecessary.

In other words, I think it's better to move the `ZeroizeOnDrop` implementation into a dedicated macro block (with checks on `core` and `buffer`) and add it to the `FixedHashTraits` list.

Reply:
I've pushed a fix for that, but now I'll need a review on https://github.com/RustCrypto/utils/pull/1192 ^^